SoftLayer/managers/dedicated_host.py

```diff
@@ -33,10 +33,138 @@ def __init__(self, client, ordering_manager=None):
         self.client = client
         self.account = client['Account']
         self.host = client['Virtual_DedicatedHost']
+        self.guest = client['Virtual_Guest']
 
         if ordering_manager is None:
             self.ordering_manager = ordering.OrderingManager(client)
 
+    def cancel_host(self, host_id):
+        """Cancel a dedicated host immediately, it fails if there are still guests in the host.
+
+        :param host_id: The ID of the dedicated host to be cancelled.
+        :return: True on success or an exception
+
+        Example::
+            # Cancels dedicated host id 12345
+            result = mgr.cancel_host(12345)
+
+        """
+        return self.host.deleteObject(id=host_id)
+
+    def cancel_guests(self, host_id):
```

Comment:
For this function, I'd like it to be able to return a structure that has a bit more information about what guests were canceled so the `slcli` can print out a receipt.

Ideally id, FQDN, and whether the guest was canceled or got an error

```
return  = [
    {'id': 1234, 'fqdn': 'test01.test.com', 'status': 'canceled' },
    {'id': 1235, 'fqdn': 'test02.test.com', 'status': 'Exception: Running Transaction or something' },
]

Reply:
Done, it returns a dictionary and it is empty if there isn't any guest in the dedicated host. The `slcli` prints a table or a message in case the dict is empty.